holoviews/plotting/mpl/chart3d.py

```diff
@@ -109,45 +109,30 @@ class Scatter3DPlot(Plot3D, PointPlot):
                                      allow_None=True, doc="""
       Index of the dimension from which the sizes will the drawn.""")
 
-    def initialize_plot(self, ranges=None):
-        axis = self.handles['axis']
-        points = self.hmap.last
-        ranges = self.compute_ranges(self.hmap, self.keys[-1], ranges)
-        ranges = match_spec(points, ranges)
-        key = self.keys[-1]
-        xs, ys, zs = (points.dimension_values(i) for i in range(3))
-
-        style = self.style[self.cyclic_index]
-        cdim = points.get_dimension(self.color_index)
-        if cdim and 'cmap' in style:
-            cs = points.dimension_values(self.color_index)
-            style['c'] = cs
-            if 'clim' not in style:
-                clims = ranges[cdim.name]
-                style.update(vmin=clims[0], vmax=clims[1])
-        if points.get_dimension(self.size_index):
-            style['s'] = self._compute_size(points, style)
-
-        scatterplot = axis.scatter(xs, ys, zs, zorder=self.zorder, **style)
-
-        self.handles['axis'].add_collection(scatterplot)
-        self.handles['artist'] = scatterplot
-
-        return self._finalize_axis(key, ranges=ranges)
-
-    def update_handles(self, axis, points, key, ranges=None):
+    def get_data(self, element, ranges, style):
+        xs, ys, zs = (element.dimension_values(i) for i in range(3))
+        self._compute_styles(element, ranges, style)
+        # Temporary fix until color handling is deterministic in py3
+        if not element.get_dimension(self.color_index):
+            style['color'] = style['c']
+        return (xs, ys, zs), style, {}
+
+    def init_artist(self, ax, plot_data, plot_kwargs):
+        scatterplot = ax.scatter(*plot_data, **plot_kwargs)
+        ax.add_collection(scatterplot)
+        return {'artist': scatterplot}
+
+    def update_handles(self, key, axis, element, ranges, style):
         artist = self.handles['artist']
-        artist._offsets3d = tuple(points[d] for d in points.dimensions())
-        cdim = points.get_dimension(self.color_index)
-        style = self.style[self.cyclic_index]
+        offsets, style, _ = self.get_data(element, ranges, style)
+        artist._offsets3d = offsets
```

Comment:
You don't seem to use ``offset`` later so I would assign it straight to ``artist._offsets3d``.